src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3544,53 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL)
+                {
+                    uint8_t scanChannel;
+                    uint16_t scanDuration;
+
+                    parsedLength = spinel_datatype_unpack(
```

Comment:
This  makes the property setter for `MAC_SCAN_STATE` to have additional arguments/values in case  of RAW APIs and if the value is `SPINEL_SCAN_STATE_ENERGY`.  I think it'd better to keep the value format of each property consistent and independent of its value. 

My suggestion is to use the same model as in the non-raw API... We can use `mScanPeriod` and/or `mScanChannelMask` here instead (or even the main channel if we want a single channel and not a channel mask for energy scan operation...).. 

Example of how to the user would issue a scan from wpantund: https://github.com/openthread/wpantund/blob/master/src/ncp-spinel/SpinelNCPTaskScan.cpp#L96